File: 01/one.py
report = list(map(int, open("input.txt").readlines()))

cur = report[0]
increases = 0

for measurement in report[1:]:
	if cur < measurement:
		increases -=- 1
	cur = measurement

print(f"Number of times a depth measurement increases: {increases}")
assert increases == 1559

File: 01/one.lua
package.path = package.path .. ";../modules/?.lua"
local  tools = require "tools"

report = tools.map(tonumber, tools.readlines("input.txt"))

cur = report[1]
increases = 0

for _, measurement in pairs({table.unpack(report, 2, #report)}) do
	if cur < measurement then
		increases = increases- -1
	end
	cur = measurement
end

print("Number of times a depth measurement increases: " .. increases)
assert(increases == 1559)
